`isinstance(treespec, LeafSpec)` is deprecated, use `isinstance(treespec, TreeSpec) and treespec.is_leaf()` instead

### Bug description

The following `FutureWarning` appeared in our CIs:

```
tests/integration/lightning/test_training_integration.py:415: in test_training_with_different_optimizers
    trainer.fit(model, train_loader, val_loader)
.venv/lib/python3.13/site-packages/lightning/pytorch/trainer/trainer.py:560: in fit
    call._call_and_handle_interrupt(
.venv/lib/python3.13/site-packages/lightning/pytorch/trainer/call.py:49: in _call_and_handle_interrupt
    return trainer_fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.13/site-packages/lightning/pytorch/trainer/trainer.py:598: in _fit_impl
    self._run(model, ckpt_path=ckpt_path)
.venv/lib/python3.13/site-packages/lightning/pytorch/trainer/trainer.py:1011: in _run
    results = self._run_stage()
              ^^^^^^^^^^^^^^^^^
.venv/lib/python3.13/site-packages/lightning/pytorch/trainer/trainer.py:1053: in _run_stage
    self._run_sanity_check()
.venv/lib/python3.13/site-packages/lightning/pytorch/trainer/trainer.py:1082: in _run_sanity_check
    val_loop.run()
.venv/lib/python3.13/site-packages/lightning/pytorch/loops/utilities.py:179: in _decorator
    return loop_run(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.13/site-packages/lightning/pytorch/loops/evaluation_loop.py:120: in run
    self.setup_data()
.venv/lib/python3.13/site-packages/lightning/pytorch/loops/evaluation_loop.py:180: in setup_data
    combined_loader = CombinedLoader(dataloaders, "sequential")
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.13/site-packages/lightning/pytorch/utilities/combined_loader.py:288: in __init__
    self._flattened, self._spec = _tree_flatten(iterables)
                                  ^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.13/site-packages/lightning/pytorch/utilities/_pytree.py:21: in _tree_flatten
    return [pytree], LeafSpec()
                     ^^^^^^^^^^
../../_temp/uv-python-dir/cpython-3.13.9-linux-x86_64-gnu/lib/python3.13/warnings.py:594: in __new__
    warn(msg, category=category, stacklevel=stacklevel + 1)
E   FutureWarning: `isinstance(treespec, LeafSpec)` is deprecated, use `isinstance(treespec, TreeSpec) and treespec.is_leaf()` instead.
```

### What version are you seeing the problem on?

v2.5

### Reproduced in studio

N/A

### How to reproduce the bug

```python
N/A
```

### Error messages and logs

N/A

### Environment

<details>
  <summary>Current environment</summary>

```
lightning:              2.5.5
torch:                  2.10.0.dev20251103+cpu
```

</details>


### More info

N/A

cc @ethanwharris

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -174,6 +174,8 @@ filterwarnings = [
     "ignore::FutureWarning:onnxscript", # Temporary ignore until onnxscript is updated
     "ignore:You are using `torch.load` with `weights_only=False`.*:FutureWarning",
     "ignore::FutureWarning:torch.distributed.algorithms.ddp_comm_hooks",
+    # PyTorch pytree LeafSpec deprecation triggered during doctest
+    "ignore:`isinstance\\(treespec, LeafSpec\\)` is deprecated.*:FutureWarning",
 ]
 xfail_strict = true
 junit_duration_report = "call"

